feat: #31_FilePage

DanielEJaimes opened the pull request:
Components:
- Dropdown
- File
- Pagination
- Sidebar
- UpdatePasswordButton 


Screens:
- FilePage

DanielEJaimes committed:
feat: #31_FilePage

Components:
- Dropdown
- File
- Pagination
- Sidebar
- UpdatePasswordButton
Screens:
- FilePage

DanielEJaimes committed:
fix: lint Pagination

PedroChaparro commented on src/main.tsx:
```diff
@@ -15,6 +16,7 @@ ReactDOM.createRoot(document.getElementById("root")!).render(
         <Route path="/" element={<App />}></Route>
         <Route path="/login" element={<Login />}></Route>
         <Route path="/register" element={<Register />}></Route>
+        <Route path="/home" element={<FilePage />}></Route>
```
Change the path to `/files`

DanielEJaimes replied:
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/4b860970-baa5-4f5f-bdec-6280021b3b6f)

PedroChaparro commented on src/components/Sidebar/Sidebar.tsx:
```diff
@@ -0,0 +1,44 @@
+      <div className="flex flex-col justify-between">
+        <div>
+          <div className="mb-2 ml-1 mr-1 mt-4">
+            <button className="w-full rounded-full bg-blue-500 px-4 py-2 text-white hover:bg-blue-700">
```
@MiguelMRojas is using `blue-600` as the main color, please, keep the main color consistent in all pages.

DanielEJaimes replied:
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/bc7172c7-925a-42ea-bd1a-822f94ac4ba7)

PedroChaparro commented on src/components/DropdownMenu/Dropdown.tsx:
```diff
@@ -0,0 +1,64 @@
+
+  return (
+    <div className="relative inline-block text-left">
+      <button
```
Since this is a button that doesn't contains any text, you should add an accessible name. Read [this](https://dequeuniversity.com/rules/axe/4.7/button-name). 

I suggest you to run `lighthouse` on your pages to detect and prevent this kind of warnings.

DanielEJaimes replied:
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/0dc2874a-e2d7-4817-9ae0-b6aa38ba1c39)

PedroChaparro commented on src/components/DropdownMenu/Dropdown.tsx:
```diff
@@ -0,0 +1,64 @@
+  return (
+    <div className="relative inline-block text-left">
+      <button
+        id="dropdownMenuIconButton"
```
Id property should be unique. Read [this](https://dequeuniversity.com/rules/axe/4.7/duplicate-id-active).

DanielEJaimes replied:
Made it unique by adding the filen ame and extension

![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/2b785506-f72a-41b0-af05-06652cb2ced5)

PedroChaparro commented on src/components/FileElement/File.tsx:
```diff
@@ -0,0 +1,32 @@
+      <div className="absolute right-0 top-0" onClick={handleDropdownClick}>
+        <Dropdown></Dropdown>
+      </div>
+      <p className="absolute bottom-0 left-1 w-full truncate">
```
There's no need to use `position: absolute` for positioning this text.

DanielEJaimes replied:
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/3f2233bd-eb76-4a1e-8dbc-915b8aef1c6f)

PedroChaparro commented on src/components/UpdatePassword/UpdatePasswordButton.tsx:
This file is not being used. You can remove it from this pull request so it can be reviewed later.

DanielEJaimes replied:
Done

PedroChaparro commented on src/screens/FilePage.tsx:
```diff
@@ -0,0 +1,53 @@
+        </div>
+        {/*Pagination*/}
+        <div className="mb-5 ml-auto mt-auto">
+          <Pagination
```
The pagination is not visible until the user scrolls down. 

Additionally, I don't think we need pagination. It wasn't requested by the client, and there are only 2-3 weeks left to finish this project, so we don't want to add extra features that could cause an overload.

DanielEJaimes replied:
Pagination removed

DanielEJaimes committed:
fix: Accessibility and style

Changed the colors to blue-600
Added the aria-label
Unique ID for each dropdownmenu

PedroChaparro committed:
Merge branch 'dev' into feat/#31_files_page

PedroChaparro commented on src/components/Sidebar/Sidebar.tsx:
```diff
@@ -0,0 +1,56 @@
+              className="w-full rounded-full bg-blue-600 px-4 py-2 text-white hover:bg-blue-700"
+              aria-label="Upload files"
+            >
+              <div className="flex items-center justify-center">
```
According to [this](https://caninclude.glitch.me/caninclude?child=div&parent=button), it's not correct to include a `div` inside a `button`. Also, it seems that the div isn't needed, you can just apply the classes directly to the `button`.

PedroChaparro commented on src/components/Sidebar/Sidebar.tsx:
```diff
@@ -0,0 +1,56 @@
+
+export function Sidebar() {
+  return (
+    <div className="col-span-2 row-span-5 h-full bg-gray-200">
```
There are a lot of `<div>` elements in this file. Since this is an aside navigation, you can use the `<aside>` tag. Also, try to remove all the unnecessary `<div>` elements.

PedroChaparro commented on src/components/Sidebar/Sidebar.tsx:
```diff
@@ -0,0 +1,56 @@
+              aria-label="Upload files"
+            >
+              <div className="flex items-center justify-center">
+                <div className="min-w-6 min-h-6">
```
You can add the classes directly to the icon instead of using a `div`. This applies to almost any icon in this file.

PedroChaparro commented on src/components/Sidebar/Sidebar.tsx:
```diff
@@ -0,0 +1,56 @@
+              </div>
+            </button>
+          </div>
+          <div className="mb-8 ml-1 mr-1 flex items-center justify-center">
```
Instead of using margins to add horizontal space (`ml-1`, `mr-1`), consider adding an horizontal padding to the `aside`.

PedroChaparro commented on src/components/Sidebar/Sidebar.tsx:
```diff
@@ -0,0 +1,56 @@
+          <div className="mb-8 ml-1 mr-1 flex items-center justify-center">
+            <button
+              className="w-full rounded-full bg-blue-600 px-4 py-2 text-white hover:bg-blue-700"
+              aria-label="New Folder"
```
The `aria-label` it's not necessary in this case, as there is a discernible text inside of the `button`.

PedroChaparro commented on src/components/Sidebar/Sidebar.tsx:
```diff
@@ -0,0 +1,56 @@
+              </div>
+            </button>
+          </div>
+          <div className="mb-4 flex cursor-pointer items-center justify-center text-blue-600 hover:text-blue-700">
```
Try to add a `gap` between the icon and the text. This also applies to the blue buttons.

PedroChaparro commented on src/screens/FilePage.tsx:
```diff
@@ -0,0 +1,36 @@
+        <div className="flex justify-start overflow-y-auto">
+          <div className="flex flex-wrap justify-start">
+            {currentFiles.map((file, index) => (
+              <div key={index} className="mb-2 mr-2 p-1">
```
This `div` is just being used to add spacing, consider adding a `gap` to the parent `div`.

PedroChaparro commented on src/screens/FilePage.tsx:
```diff
@@ -0,0 +1,36 @@
+          />
+        </div>
+        <div className="flex justify-start overflow-y-auto">
+          <div className="flex flex-wrap justify-start">
```
You can just add the `flex-wrap` and `justify-start` to the parent `div` to get rid of this extra `div`.

PedroChaparro commented on src/components/FileElement/File.tsx:
```diff
@@ -0,0 +1,32 @@
+      className="relative left-2 top-2 box-border flex h-32 w-32 cursor-pointer flex-col rounded-lg border-2 bg-white p-5"
+      onClick={handleClick}
+    >
+      <FileText className="h-20 w-20" />
```
Should display a directory icon if the file is a directory.

PedroChaparro commented on src/components/FileElement/File.tsx:
```diff
@@ -0,0 +1,32 @@
+
+  return (
+    <div
+      className="relative left-2 top-2 box-border flex h-32 w-32 cursor-pointer flex-col rounded-lg border-2 bg-white p-5"
```
Avoid using position to add spacing (`left-2`, `top-2`). Consider adding padding to the parent container.

PedroChaparro commented on src/screens/FilePage.tsx:
```diff
@@ -0,0 +1,36 @@
+            className="mt-2 w-full rounded-md border border-gray-300 px-4 py-2"
+          />
+        </div>
+        <div className="flex justify-start overflow-y-auto">
```
The `overflow-y-auto` is causing the following bug: 

<image src="https://github.com/hawks-atlanta/frontend-react/assets/62714297/973b073e-2511-480d-af64-d9349a6f66a4" width="350px" />

DanielEJaimes committed:
fix: requested changes

DanielEJaimes committed:
revert: protect route

PedroChaparro committed:
refactor: Remove unnecessary div and add comments

PedroChaparro committed:
fix(ui): Set fixed height to the navbar

PedroChaparro committed:
refactor(ui): Update file component

PedroChaparro committed:
refactor: Remove unnecessary div

PedroChaparro committed:
feat(ui): Add hover effect to file component

PedroChaparro reviewed (approved):
Squash and merge :octocat::

![image](https://github.com/hawks-atlanta/frontend-react/assets/62714297/30c3ecc6-7457-44c0-bc0e-38ff31a77d19)

Remember that you need to fetch the Python Proxy to obtain the actual user's files. You can do this in a separate Pull Request, with the corresponding tests.
